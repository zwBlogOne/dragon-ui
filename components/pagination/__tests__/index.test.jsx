import React from 'react';
import { render, mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import Pagination from '../index';

describe('Pagination', () => {
  it('renders normal Pagination correctly', () => {
    const wrapper = render(
      <div>
        <Pagination total={100} />
      </div>
    );

    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('renders Pagination with total info correctly', () => {
    const wrapper = render(
      <div>
        <Pagination total={100} showTotal />
      </div>
    );

    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('renders Pagination with jumper correctly', () => {
    const wrapper = render(
      <div>
        <Pagination total={100} showJumper />
      </div>
    );

    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('renders Pagination with customized props correctly', () => {
    const wrapper = render(
      <div>
        <Pagination
          total={100}
          radius
          bordered
          addonBefore="addonBefore"
          addonAfter="addonAfter"
        />
      </div>
    );

    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('behaves correctly when changing page', () => {
    const onChange = jest.fn();
    const wrapper = mount(
      <div>
        <Pagination total={100} onPageChange={onChange} />
      </div>
    );

    wrapper.find('.ui-pagination-item').at(2).simulate('click');

    expect(onChange).toHaveBeenCalledWith(2);
  });

  it('behaves correctly when click prev button', () => {
    const onChange = jest.fn();
    const wrapper = mount(
      <div>
        <Pagination total={100} value={6} onPageChange={onChange} />
      </div>
    );

    wrapper.find('.ui-pagination-item-prev').simulate('click');

    expect(onChange).toHaveBeenCalledWith(5);
  });

  it('behaves correctly when click next button', () => {
    const onChange = jest.fn();
    const wrapper = mount(
      <div>
        <Pagination total={100} value={5} onPageChange={onChange} />
      </div>
    );

    wrapper.find('.ui-pagination-item-next').simulate('click');

    expect(onChange).toHaveBeenCalledWith(6);
  });

  it('behaves correctly when click prev 5 button', () => {
    const onChange = jest.fn();
    const wrapper = mount(
      <div>
        <Pagination total={100} value={6} onPageChange={onChange} />
      </div>
    );

    wrapper.find('.ui-pagination-item-jump-prev').simulate('click');

    expect(onChange).toHaveBeenCalledWith(1);
  });

  it('behaves correctly when click next 5 button', () => {
    const onChange = jest.fn();
    const wrapper = mount(
      <div>
        <Pagination total={100} value={1} onPageChange={onChange} />
      </div>
    );

    wrapper.find('.ui-pagination-item-jump-next').simulate('click');

    expect(onChange).toHaveBeenCalledWith(6);
  });

  it('behaves correctly when change page with jumper', () => {
    const onChange = jest.fn();
    const wrapper = mount(
      <div>
        <Pagination showJumper total={100} value={1} onPageChange={onChange} />
      </div>
    );

    wrapper.find('.ui-pagination-jumper input').simulate('keydown', {
      keyCode: 13,
      target: {
        value: 6,
      },
    });
    expect(onChange).toHaveBeenCalledWith(6);
  });

  it('behaves correctly when receiving new value', () => {
    const wrapper = mount(
      <Pagination showJumper total={100} value={2} />
    );

    wrapper.setProps({ value: 1 });
    expect(wrapper.find('.ui-pagination-item').at(1).hasClass('ui-pagination-item-active')).toBeTruthy();
  });
});
