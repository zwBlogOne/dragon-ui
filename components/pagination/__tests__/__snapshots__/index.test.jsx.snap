// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pagination renders Pagination with customized props correctly 1`] = `
<div>
  <div
    class="ui-pagination bordered radius"
  >
    <div
      class="ui-pagination-pagers"
    >
      <ul>
        <div
          class="ui-pagination-addon-before"
        >
          addonBefore
        </div>
        <li
          class="ui-pagination-item ui-pagination-item-prev ui-pagination-item-disabled"
          title="上一页"
        >
          <i
            class="ui-icon ui-icon-arrow-left theme-default"
          />
        </li>
        <li
          class="ui-pagination-item ui-pagination-item-active"
          title="1"
        >
          1
        </li>
        <li
          class="ui-pagination-item"
          title="2"
        >
          2
        </li>
        <li
          class="ui-pagination-item"
          title="3"
        >
          3
        </li>
        <li
          class="ui-pagination-item"
          title="4"
        >
          4
        </li>
        <li
          class="ui-pagination-item"
          title="5"
        >
          5
        </li>
        <li
          class="ui-pagination-item ui-pagination-item-jump-next"
          title="向后 5 页"
        />
        <li
          class="ui-pagination-item"
          title="最后一页"
        >
          10
        </li>
        <li
          class="ui-pagination-item ui-pagination-item-next"
          title="下一页"
        >
          <i
            class="ui-icon ui-icon-arrow-right theme-default"
          />
        </li>
        <div
          class="ui-pagination-addon-after"
        >
          addonAfter
        </div>
      </ul>
    </div>
  </div>
</div>
`;

exports[`Pagination renders Pagination with jumper correctly 1`] = `
<div>
  <div
    class="ui-pagination"
  >
    <div
      class="ui-pagination-pagers"
    >
      <ul>
        <li
          class="ui-pagination-item ui-pagination-item-prev ui-pagination-item-disabled"
          title="上一页"
        >
          <i
            class="ui-icon ui-icon-arrow-left theme-default"
          />
        </li>
        <li
          class="ui-pagination-item ui-pagination-item-active"
          title="1"
        >
          1
        </li>
        <li
          class="ui-pagination-item"
          title="2"
        >
          2
        </li>
        <li
          class="ui-pagination-item"
          title="3"
        >
          3
        </li>
        <li
          class="ui-pagination-item"
          title="4"
        >
          4
        </li>
        <li
          class="ui-pagination-item"
          title="5"
        >
          5
        </li>
        <li
          class="ui-pagination-item ui-pagination-item-jump-next"
          title="向后 5 页"
        />
        <li
          class="ui-pagination-item"
          title="最后一页"
        >
          10
        </li>
        <li
          class="ui-pagination-item ui-pagination-item-next"
          title="下一页"
        >
          <i
            class="ui-icon ui-icon-arrow-right theme-default"
          />
        </li>
      </ul>
      <div
        class="ui-pagination-jumper"
      >
        跳至
        <span>
          <input
            class="ui-input size-sm"
            style="width:50px;text-align:center;margin-left:5px;margin-right:5px"
            type="text"
            value=""
          />
        </span>
        页
      </div>
    </div>
  </div>
</div>
`;

exports[`Pagination renders Pagination with total info correctly 1`] = `
<div>
  <div
    class="ui-pagination"
  >
    <div
      class="ui-pagination-total"
    >
      共有 100 条记录 第 1 / 10 页
    </div>
    <div
      class="ui-pagination-pagers"
    >
      <ul>
        <li
          class="ui-pagination-item ui-pagination-item-prev ui-pagination-item-disabled"
          title="上一页"
        >
          <i
            class="ui-icon ui-icon-arrow-left theme-default"
          />
        </li>
        <li
          class="ui-pagination-item ui-pagination-item-active"
          title="1"
        >
          1
        </li>
        <li
          class="ui-pagination-item"
          title="2"
        >
          2
        </li>
        <li
          class="ui-pagination-item"
          title="3"
        >
          3
        </li>
        <li
          class="ui-pagination-item"
          title="4"
        >
          4
        </li>
        <li
          class="ui-pagination-item"
          title="5"
        >
          5
        </li>
        <li
          class="ui-pagination-item ui-pagination-item-jump-next"
          title="向后 5 页"
        />
        <li
          class="ui-pagination-item"
          title="最后一页"
        >
          10
        </li>
        <li
          class="ui-pagination-item ui-pagination-item-next"
          title="下一页"
        >
          <i
            class="ui-icon ui-icon-arrow-right theme-default"
          />
        </li>
      </ul>
    </div>
  </div>
</div>
`;

exports[`Pagination renders normal Pagination correctly 1`] = `
<div>
  <div
    class="ui-pagination"
  >
    <div
      class="ui-pagination-pagers"
    >
      <ul>
        <li
          class="ui-pagination-item ui-pagination-item-prev ui-pagination-item-disabled"
          title="上一页"
        >
          <i
            class="ui-icon ui-icon-arrow-left theme-default"
          />
        </li>
        <li
          class="ui-pagination-item ui-pagination-item-active"
          title="1"
        >
          1
        </li>
        <li
          class="ui-pagination-item"
          title="2"
        >
          2
        </li>
        <li
          class="ui-pagination-item"
          title="3"
        >
          3
        </li>
        <li
          class="ui-pagination-item"
          title="4"
        >
          4
        </li>
        <li
          class="ui-pagination-item"
          title="5"
        >
          5
        </li>
        <li
          class="ui-pagination-item ui-pagination-item-jump-next"
          title="向后 5 页"
        />
        <li
          class="ui-pagination-item"
          title="最后一页"
        >
          10
        </li>
        <li
          class="ui-pagination-item ui-pagination-item-next"
          title="下一页"
        >
          <i
            class="ui-icon ui-icon-arrow-right theme-default"
          />
        </li>
      </ul>
    </div>
  </div>
</div>
`;
